montblanc/impl/biro/v2/loaders/loaders.py

```diff
@@ -59,8 +82,11 @@ def load(self, solver, **kwargs):
         solver.set_ref_wave(montblanc.constants.C/tf.getcol('REF_FREQUENCY')[0])
 
         # Get the baseline antenna pairs and correct the axes
-        ant1 = tm.getcol('ANTENNA1').reshape(ntime,nbl)
-        ant2 = tm.getcol('ANTENNA2').reshape(ntime,nbl)
+        ant1 = tm.getcol('ANTENNA1').reshape(file_ant_shape)
+        ant2 = tm.getcol('ANTENNA2').reshape(file_ant_shape)
+	if (data_order == ORDER_BL_TIME):
+		ant1.transpose(ant_transpose)
+		ant2.transpose(ant_transpose)
```

Comment:
But a method is not always something that modifies an object's state. A method that returns something (e.g. a new instance) based on the state of the object is also perfectly valid OOP. Whether a method modifies an object or not is just part of that method's semantics (and in C++ this can be enforced by declaring methods as const). A method that modifies an object in place is also perfectly pythonic (c.f. list.sort()).

numpy.array.transpose() is a "const method" in C++ terms, and is defined as returning a new array instance that is a transposed view of the original array. You could also say:

```
ant1t = ant1.transpose(ant_transpose)
```

...and end up with two array objects (that actually share the same underlying array storage, hence "views" of the array). The assignment back to ant1 is only necessary because you want to throw away the non-transposed view of the array.

The numpy API tends to avoid in-place modification. This is by choice, since it encourages the use of "array views", which is actually very powerful and leads to very efficient code, if used right.